client/dashboard/sites/staging-site-sync-modal/index.tsx

```diff
@@ -0,0 +1,275 @@
+import { useQuery } from '@tanstack/react-query';
```

Comment:
I have used `client/sites/staging-site/components/staging-site-sync-modal/index.tsx` as a base for this new component and removed mostly parts related to the `FileBrowser` component we won't need.